### PR TITLE
feat(ui): toggle component

### PR DESCRIPTION
**Pull request by 32penkin**
### Please read and mark the following check list before creating a pull request:

 - [] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/react-native-ui-kitten/blob/master/CONTRIBUTING.md) guide.
 - [] I read and followed the [New Feature Checklist](https://github.com/akveo/react-native-ui-kitten/blob/master/DEV_DOCS.md) guide.

 #### Short description of what this resolves:

### CODE REVIEW

File: src/framework/ui/index.ts [hunk 1]
```diff
@@ -7,14 +7,18 @@ import {
   RadioGroup as RadioGroupComponent,
   Props as RadioGroupProps,
 } from './radioGroup/radioGroup.component';
+import {
+  Toggle,
```
Comment (artyorsh) [inline src/framework/ui/index.ts]: The same as in [Layout](https://github.com/akveo/react-native-ui-kitten/pull/240#discussion_r249254909)

File: src/framework/ui/toggle/toggle.component.tsx [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+  onValueChange?: (value: boolean) => void;
+  value?: boolean;
+  appearance?: string | 'default';
+  status?: string | 'error';
+  size?: string | 'big' | 'small';
+}
+
+export type ToggleProps = ToggleComponentProps & StyledComponentProps & ViewProps;
+
+export class Toggle extends React.Component<ToggleProps> {
```
Comment (artyorsh) [inline src/framework/ui/toggle/toggle.component.tsx]: [relevant code style issue](https://github.com/akveo/react-native-ui-kitten/pull/240#discussion_r249254932)

File: src/framework/ui/toggle/toggle.component.tsx [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+      onMoveShouldSetPanResponderCapture: this.onMoveShouldSetPanResponderCapture,
+      onPanResponderTerminationRequest: this.onPanResponderTerminationRequest,
+      onPanResponderGrant: () => this.onPanResponderGrant(thumbSize),
+      onPanResponderMove: this.onPanResponderMove,
+      onPanResponderRelease: (evt: GestureResponderEvent, gestureState: PanResponderGestureState) =>
+        this.onPanResponderRelease(evt, gestureState, thumbSize),
+    });
+  }
+
+  animateSwitch = (value: boolean, callback: () => void = () => null) => {
```
Comment (artyorsh) [inline src/framework/ui/toggle/toggle.component.tsx]: can be private?

File: src/framework/ui/toggle/toggle.component.tsx [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+              size={componentStyle.thumbComponentSize}
+              color={interpolatedCheckColor}
+              isAnimated={true}
+            />
+          </Animated.View>
+        </Animated.View>
+      </View>
+    );
+  }
+
```
Comment (artyorsh) [inline src/framework/ui/toggle/toggle.component.tsx]: [i'm your linter :)](https://github.com/akveo/react-native-ui-kitten/pull/240#discussion_r249255049)

File: src/framework/ui/toggle/toggle.spec.config.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
+export const mapping = {
```
Comment (artyorsh) [inline src/framework/ui/toggle/toggle.spec.config.ts]: property type is missing

File: src/framework/ui/toggle/toggle.component.tsx [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+} from '@kitten/theme';
+import { CheckMark } from '../common/checkmark.component';
+
+interface ToggleComponentProps {
+  disabled?: boolean;
+  onValueChange?: (value: boolean) => void;
+  value?: boolean;
+  appearance?: string | 'default';
+  status?: string | 'error';
+  size?: string | 'big' | 'small';
```
Comment (malashkevich) [inline src/framework/ui/toggle/toggle.component.tsx]: please remove exact values

File: src/framework/ui/toggle/toggle.spec.config.ts [hunk 7]
```diff
@@ -0,0 +1,92 @@
+import { ThemeMappingType } from 'eva/packages/common';
```
Comment (artyorsh) [inline src/framework/ui/toggle/toggle.spec.config.ts]: linter issues across whole file

**Pull request merged**